### PR TITLE
Fix TTT Predicted Viewmodels and TTTPlayerSpeedMultiplier > 1

### PR DESCRIPTION
**Pull request by meepdarknessmeep**

### CODE REVIEW

**Comment by meepdarknessmeep** [comment]
I also just added a `slowed` and `CMoveData mv` argument to TTTPlayerSpeedModifier to allow greater changes within the hook. Adding more functionality probably wouldn't be bad, and I hope you agree.

**Comment by Heyter** [comment]
if player crouching then movespeed = 220.
try
`		if IsValid(wep) and wep.GetIronsights and wep:GetIronsights() then
			basespeed = 120
		elseif ply:Crouching() then
			basespeed = 66
		end`

**Comment by meepdarknessmeep** [comment]
I will fix it normally, hold on.

**Comment by mcNuggets1** [comment]
Should be merged ASAP.

File: garrysmod/gamemodes/terrortown/gamemode/shared.lua [hunk 1]
```diff
@@ -167,17 +167,18 @@ end
...
 function GM:Move(ply, mv)
    if ply:IsTerror() then
-      local basespeed = 220
+      local basemul = 1
       local slowed = false
       -- Slow down ironsighters
       local wep = ply:GetActiveWeapon()
       if IsValid(wep) and wep.GetIronsights and wep:GetIronsights() then
-         basespeed = 120
+         basespeed = 120 / 220
```
Comment (mcNuggets1) [inline garrysmod/gamemodes/terrortown/gamemode/shared.lua]: Should this remain basespeed, when everything is changed to basemul?

**Comment by mcNuggets1** [comment]
And for some reason when running the viewmodel stutters.

**Comment by meepdarknessmeep** [comment]
I'm not sure how I missed that, see anything else @mcNuggets1?

**Comment by mcNuggets1** [comment]
Not really. Everything should be fine.
I didn't test the gun base as I got my own one, which works better :P.

**Comment by Kefta** [comment]
That kind of defeats the purpose of testing this PR.

**Comment by mcNuggets1** [comment]
I am not responsible for testing this PR.

If meepdarknessmeep wants, he could ask if he would get a copy of my ttt gun base to compare.
But to be honest I don't think aiming in TTT ever played a role, besides using a sniper.

**Comment by meepdarknessmeep** [comment]
@svdm i don't think anyone's had issues with it and everything has been tested by me, so it might be a good time to pull this.

**Comment by meepdarknessmeep** [comment]
@svdm can you add the CMoveData argument to TTTPlayerSpeedModifier on the TTT wiki?

**Pull request merged**